### ISSUE TITLE
Running SIMBA3D on multiple CPU cores?

### ISSUE THREAD

**Issue by tarak77** [nerettilab/SIMBA3D#5]
Hi again Michael,

SIMBA3D works good but is it possible to save time by assigning multiple CPU cores to work with? I have a machine with Xeon CPUs can run 16+ processes. 

-Tarak

**Comment by michael-m-rosenthal** [comment]
Yes, it uses the multiprocessing package to complete independent tasks in
embarrassingly parallel. This doesn't work for warmstart subtasks because
the tasks need to run sequentially.

The -c option should work well on desktops and laptops if you are running
independent  tasks on different data sets.

I have found that the multiprocessing package does not work well in HPC
settings and it was way easier just to run multiple instances on individual
tasks than to wait in que trying to debug how to get it to run correctly.

Also, I have found that the multiprocessing package hides error messages,
which can make debugging a nightmare.

On Aug 16, 2018 3:07 PM, "Tarak Shisode" <notifications@github.com> wrote:

Hi again Michael,

SIMBA3D works good but is it possible to save time by assigning multiple
CPU cores to work with? I have a machine with Xeon CPUs can run 16+
processes.

-Tarak

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/nerettilab/SIMBA3D/issues/5>, or mute the thread
